src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from "react";
+import { VenueAccessMode } from "types/VenueAcccess";
+import { AnyVenue } from "types/venues";
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { joinVenue } from "utils/url";
+import { SecretPasswordForm } from "../SecretPasswordForm";
+
+interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit: () => void;
+  onPasswordSuccess: () => void;
+}
+
+export const VenueJoinButton: React.FC<VenueJoinButtonProps> = ({
+  venueId,
+  venue,
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, venue?.entrance);
+  }, [venue, venueId]);
+
+  // @debt Handle emails and codes as well.
+  if (venue.access === VenueAccessMode.Password) {
+    return (
+      <div className="secret-password-form-wrapper">
+        <SecretPasswordForm
+          buttonText={
+            venue.config?.landingPageConfig.joinButtonText ?? "Join the party"
+          }
+          onPasswordSubmit={onPasswordSubmit}
+          onPasswordSuccess={onPasswordSuccess}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <button
+      className="btn btn-primary btn-block btn-centered"
+      onClick={handleJoinVenue}
+    >
+      Join the event
+      {(venue?.start_utc_seconds ?? 0) > getCurrentTimeInUTCSeconds() && (
+        <span className="countdown">
+          Begins in {getTimeBeforeParty(venue?.start_utc_seconds)}
+        </span>
+      )}
```

Comment:
Do we want to show this countdown regardless of the Venue Access protection method?

Reply:
Good question, I don't have enough info to answer though. Ed is the one who knows more about this. I'll ask tomorrow on the daily.

Reply:
Sounds good.

Reply:
This is supposed to be visible for all of the venue access types. I spoke with Ed about this yesterday.

Reply:
Cool. Then will probably require some changes here to ensure that (as from memory, it was only shown on the 'final branch' of the conditional renders.